### ISSUE TITLE
[0.65.0] npx react-native init AwesomeProject ERESOLVE dependency failure

### ISSUE THREAD

**Issue by fondakogb** [facebook/react-native#32049]
## Description
When I try to create a fresh/new project as for documentation with following command:

`npx react-native init AwesomeProject`

the execution fail with following error(s):

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: AwesomeProject@0.0.1
npm ERR! Found: react@17.0.2
npm ERR! node_modules/react
npm ERR!   react@"17.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"17.0.1" from react-test-renderer@17.0.1
npm ERR! node_modules/react-test-renderer
npm ERR!   dev react-test-renderer@"17.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See ~/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2021-08-19T12_56_21_294Z-debug.log
(node:12013) UnhandledPromiseRejectionWarning: Error: Command failed: npm install --save --save-exact react@17.0.2
    at makeError (~/rnprojects/AwesomeProject/node_modules/execa/index.js:174:9)
    at ~/rnprojects/AlefProject/node_modules/execa/index.js:278:16
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async generateProject (~/rnprojects/AlefProject/node_modules/@react-native-community/cli/build/commands/init/initCompat.js:122:3)
    at async Object.initCompat (~/rnprojects/AlefProject/node_modules/@react-native-community/cli/build/commands/init/initCompat.js:105:3)
```

## React Native version:
```
System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i5-3210M CPU @ 2.50GHz
    Memory: 577.77 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.17.5 - /usr/local/opt/node@14/bin/node
    Yarn: Not Found
    npm: 6.14.14 - /usr/local/opt/node@14/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - ~/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 30
      Build Tools: 30.0.3
      System Images: android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7199119
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_282 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.65.0 => 0.65.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
**Note I tried also with:**

    Node: 16.7.0 - /usr/local/opt/node@16/bin/node
    npm: 7.20.3 - /usr/local/opt/node@16/bin/npm
obtaining same (error) result

## Steps To Reproduce

1. open terminal
2. digit `npx react-native init AwesomeProject` and press `enter`
3. (obtain the above error)

## Expected Results
It should be possible to successfully create a fresh react native app-project (from default template) in the expected (sub)directory AwesomeProject.

**Comment by DoctorJohn** [comment]
@fondakogb This has been fixed in [0.65.1](https://github.com/facebook/react-native/releases/tag/v0.65.1) which was released a few hours ago.

> Set react-test-renderer to 17.0.2 in the template

**Comment by fondakogb** [comment]
@DoctorJohn I confirm that this dependency-issue has been fixed in [0.65.1](https://github.com/facebook/react-native/releases/tag/v0.65.1).
Thank you all for your great work
